include/libgit4cpp/GitRepository.h

```diff
@@ -79,7 +79,7 @@ class GitRepository
     git_repository* get_repo();
 
     /// Stage all new and changed files and folders in the repository directory.
-    void add();
+    void add(const std::string& glob = "*");
```

Comment:
I think the "glob" parameter needs some explanation. Can I set it to "*.txt" to stage all new/changed text files instead of all?

Reply:
... and may `glob` be empty?

Reply:
>  Can I set it to "*.txt"

I don't know, probably. Setting it is rather easy ;-D

> stage all new/changed text files instead of all

That's how it is supposed to work, although I only tested globs ending in a star.
The upstream documentation is ... nonexisting (or I am too dump).

![image](https://github.com/taskolib/libgit4cpp/assets/16012374/8a0e7dfd-3e94-4f89-ae59-4dc176478fb5)

Without reverse-engineering I can not say more than the few words given above.

Reply:
Added docu and a lot of tests that show hopefully everything.

![image](https://github.com/taskolib/libgit4cpp/assets/16012374/a8980011-7d5a-4c4c-8e5a-cc92a460e70f)